src/app/websites/website-services/afterdark/afterdark.service.ts

```diff
@@ -0,0 +1,201 @@
+import { LoginProfileManagerService } from 'src/app/login/services/login-profile-manager.service';
+import { fileAsFormDataObject } from 'src/app/utils/helpers/file.helper';
+import { TypeOfSubmission } from 'src/app/utils/enums/type-of-submission.enum';
+import { Folder } from '../../interfaces/folder.interface';
+
+const ACCEPTED_FILES = ['png', 'jpeg', 'jpg'];
+
+function submissionValidate(submission: Submission, formData: SubmissionFormData): any[] {
+  var tags = [];
+  if (formData.defaults && formData.defaults.tags) {
+    tags = formData.defaults.tags.tags;
+  }
+  if (tags.length < 2) {
+    return [['AfterDark requires at least two tags', { website: 'AfterDark', value: tags.length }]];
+  }
+  const collections = ((formData['AfterDark'] || {}).options || {}).folders || [];
+  if (collections.length < 1) {
+    return [['AfterDark requires at least one collection folder', { website: 'AfterDark', value: collections.length }]];
+  }
+  return [];
+}
+
+@Injectable({
+  providedIn: 'root'
+})
+@Website({
+  acceptedFiles: ACCEPTED_FILES,
+  additionalFiles: false,
+  displayedName: 'AfterDark',
+  login: {
+    url: 'https://afterdark.art/login',
+  },
+  components: {
+    submissionForm: AfterDarkSubmissionForm,
+  },
+  validators: {
+    submission: submissionValidate
+  },
+  preparsers: {},
+  parsers: {
+    description: [(html: string) => html],
+  },
+})
+export class AfterDark extends BaseWebsiteService implements WebsiteService {
+  readonly BASE_URL: string = 'https://afterdark.art';
+
+  constructor(private _profileManager: LoginProfileManagerService) {
+    super();
+  }
+
+  private async getCsrfToken(profileId: string): Promise<string> {
+    var cookies = await getCookies(profileId, this.BASE_URL);
+    const response = await got.get(this.BASE_URL, this.BASE_URL, cookies, profileId);
+    if (response.statusCode !== 200) {
+      throw new Error(`Bad response getting csrf token ${response.statusCode}`);
+    }
+    const match = response.body.match(new RegExp(`name="csrfmiddlewaretoken" value="([^"]+)"`));
+    if (!match) {
+      throw new Error('Could not find csrf token in body');
+    }
+    const csrfToken = match[1];
+    return csrfToken;
+  }
+
+  private async getUserData(profileId: string, csrfToken: string): Promise<any> {
+    const body = JSON.stringify({
+      "operationName": "UserQuery",
+      "variables": {},
+      "query": `query UserQuery {
+        userViewer {
+          user {
+            id
+            username
+            isUserAdmin
+            isProUser
+            isEmailVerified
+            collectionSet {
+              edges {
+                node {
+                  id
+                  title
+                }
+              }
+            }
+          }
+        }
+      }`
+    });
+    var cookies = await getCookies(profileId, this.BASE_URL);
+    const response = await got.gotPostJSON(`${this.BASE_URL}/graphql/`, body, this.BASE_URL, cookies, profileId, {
```

Comment:
I am like 99% certain a new get.gotPostJSON did not need to be made.
Both Request and Got should be fine with sending non-formdata.
`await got.post(`${this.BASE_URL}/graphql/`, null, this.BASE_URL, cookies, profileId, {
    body,
    json: true,
    headers: {...}
})`

https://github.com/request/request#requestoptions-callback

Reply:
Ahh, I was thrown by having to set the second param to `{}`.  Setting it to null is an error because gotPost doesn't check if the body is null.  Removed getPostJSON.